### PR TITLE
Saurian Race Description

### LINKED ISSUE #3372: Adding info for Saurians in the Help
There is more info avaible in the [wesnoth wiki](https://wiki.wesnoth.org/Category:Races) than in the ingame Help regarding the [Saurians](https://wiki.wesnoth.org/Saurians)~, [Drakes](https://wiki.wesnoth.org/Drakes_(race)), [Elves](https://wiki.wesnoth.org/Elves), [Dwarves](https://wiki.wesnoth.org/Dwarves) and [Monsters](https://wiki.wesnoth.org/Races#Monsters)~. 
It would be really nice to add those informations to the ingame Help. :)
(I do not know if/which parts of the wiki are intentionally not in the game help.)

~EDIT: While I'm at it, could we maybe add an 'Outlaw' section to the 'Humans'? I feel that they are very much missing in the description.~
(Moved to a new issue.)

### PR DESCRIPTION
**Pull request by sigurdfdragon**
Fixes #3372  (fix for #3392 to be later for master only)

Gives the Saurians a race description (taken from the wiki, with the minor change of comparing height against humans instead of drakes) The wiki description has been around for several years.

Has the Saurian warrior line be female, in accordance with the new race description, some lore in SotA and some UMC campaigns, which have likely gotten this from the wiki.

### CODE REVIEW

File: data/core/units.cfg [hunk 1]
```diff
@@ -247,7 +247,24 @@ The Dunefolk’s inquisitive and explorative nature does not preclude military s
         male_name= _ "race^Saurian"
         female_name= _ "race+female^Saurian"
         plural_name= _ "race^Saurians"
-        description= "" # wmllint: ignore
+        description= "Saurians are small, slender, and lizard-like creatures. Much smaller than humans, they rarely stand taller than a ten year old child, though the distance from the tip of their snout to the end their tail can be as long as the average man is tall. Light and nimble, the warriors prefer to fight as they hunt- slipping through enemy lines to target the weak and the injured while their slender figures allow them to evade their attackers.
```
Comment (jostephd) [inline data/core/units.cfg]: The hyphen in "as they hunt-" should be an em dash (&mdash;) or a colon.

File: data/core/units.cfg [hunk 2]
```diff
@@ -247,7 +247,24 @@ The Dunefolk’s inquisitive and explorative nature does not preclude military s
...
+
+<header>text='Society'</header>
+Saurians are very mysterious creatures due to their tendency to live in areas inhospitable to others, such as swamps. Fatalistic in the extreme, Saurians believe all the events in a life can be predicted by the use of a complex form of astrology.
+
+Saurian culture is sharply segregated between the genders. Within each gender the members compete and through skill, determination, and reputation establish a clear pecking order, with a chief at the top. On those occasions when the two genders interact they do not contest for dominance, the situation determines the dominant gender. The chief of the males is alpha if they are within their village or encampment while the chief of females is dominant anywhere else. This continues down the rank structure with each male or female being dominant over any member of the opposite gender with lower rank and submitting to members of the opposite gender with higher rank.
```
Comment (jostephd) [inline data/core/units.cfg]: Comma splice at "dominance,".

"if they are within their" - ambiguous pronoun (could refer either to the males or to the tribe). "...is alpha within the clan's village or encampment..." maybe?

File: data/core/units.cfg [hunk 3]
```diff
@@ -247,7 +247,24 @@ The Dunefolk’s inquisitive and explorative nature does not preclude military s
...
+
+The segregation and alternating gender-dominance of Saurian society is an outgrowth of their clearly defined gender roles. It is the responsibility of the female to hunt and find food, skills which ultimately train them to be warriors; the skirmishers, flankers, and ambushers other races so fear. Males, meanwhile, are responsible for guarding the clutch- the eggs left by the females. While this leaves time for the males to develop and hone the arts of astrology, healing, and magic, it also exposes them to significant danger, as they are stationary targets for a Saurian clan's number one enemy- other Saurian clans.
```
Comment (jostephd) [inline data/core/units.cfg]: The hyphens should become em dashes or colons.

File: data/core/units.cfg [hunk 4]
```diff
@@ -247,7 +247,24 @@ The Dunefolk’s inquisitive and explorative nature does not preclude military s
...
+
+New Saurian clans are started when the proper astrological signs are read. Called a “hatching,” the female indicated by the conjunction select a group of individuals with lower rank and leave their source clan. Frequently all females with a specific trait will be selected, causing multiple clans to “hatch” at the same time. Selection is a simple process, no group leaving can be larger than any other, all the groups together cannot be larger, than the group being left, and higher ranking allows a female to overrule another female's choice of who they take.
```
Comment (jostephd) [inline data/core/units.cfg]: "the female indicated... select" -> "each female indicated... selects"

"selected" -> "indicated"

"process," -> "process:" and remove the comma before "than"

**Comment by sevu** [comment]
I am not sure if the gender change can go into 1.14, I could imagine that it may potentially cause OOS with the charm ability form ageless.

File: data/core/units.cfg (hunk 1 above)
Comment (CelticMinstrel) [reply data/core/units.cfg]: "as they hunt: slipping" (colon)

or 

"as they hunt—slipping" (em dash)

or

"as they hunt – slipping" (en dash)

I'm not sure if Wesnoth has a guide about whether to prefer the em or en dash.

Comment (CelticMinstrel) [reply data/core/units.cfg]: Also, I think I'd prefer "small, slender, lizard-like creatures", without the explicit conjunction. Though the "small" is also redundant with the next sentence... and "slender" is also repeated later on... maybe the first sentence should be rephrased or even mostly-removed?

"to the end of their tail" - missing a preposition

**Comment by sigurdfdragon** [comment]
@jostephd Thanks, updated. Is the comma splice fix good?

File: data/core/units.cfg (hunk 2 above)
Comment (CelticMinstrel) [reply data/core/units.cfg]: "compete and through skill..." seems weird. I'd add a comma or semicolon before the "and" here to make it more clear that we're starting a new clause.

File: data/core/units.cfg [hunk 5]
```diff
@@ -247,7 +247,24 @@ The Dunefolk’s inquisitive and explorative nature does not preclude military s
...
+
+Because of the rapid growth of saurian populations, these frequent splits in clans, and the fact that cannibalism is not taboo among Saurians violence is one of the defining features of the Saurian life. This limits the growth of the Saurian culture, limiting it to fits and starts- as much of their knowledge is passed by oral tradition and their possessions must be mobile.
```
Comment (CelticMinstrel) [inline data/core/units.cfg]: Missing a comma before "violence"? Also drop this entire part: ", limiting it " and replace that hyphen with a comma.

**Comment by CelticMinstrel** [comment]
You don't need to add `gender=male` to the mages?

**Comment by jostephd** [comment]
@sigurdfdragon Welcome. I'd change ", instead," to "; instead,".

File: data/core/units.cfg (hunk 1 above)
Comment (jostephd) [reply data/core/units.cfg]: > I'm not sure if Wesnoth has a guide about whether to prefer the em or en dash.

data/campaigns/ has plenty of uses of em dash and none of en dash.

File: data/core/units.cfg (hunk 2 above)
Comment (jostephd) [reply data/core/units.cfg]: That sentence sounds good to me. In everyday English I'd have put the verb "establish" before its modifier phrase "through skill, determination, and reputation", but Wesnoth doesn't tend to use everyday English :P

File: data/core/units.cfg [hunk 6]
```diff
@@ -247,7 +247,24 @@ The Dunefolk’s inquisitive and explorative nature does not preclude military s
...
+
+<header>text='Geography'</header>
+Saurians can live in many different areas, though swamps are by far their most common habitat.
+
+<header>text='Biology'</header>
+Saurians live spectacularly short lives by comparison to most of the other races of Wesnoth, reaching full adulthood within three years and often dying by the time they are 10 to 15 years old. By far, the most common cause of death is violence. Saurian females produce clutches of about 20 eggs roughly once a year, which creates constant population pressure and would stress most carnivores' food supply. Hunters and scavengers, Saurians have extremely strong jaws and have a very powerful digestive system with highly acidic fluids, making them capable of eating and digesting their entire prey including skin, teeth, horns and bones. Further they have no aversion to and readily eat carrion and commit cannibalism. Due to this, the males of the Saurian race have specialized in cold-magics to take advantage of their own cold-blooded nature."
```
Comment (jostephd) [inline data/core/units.cfg]: Sorry, missed this paragraph in previous review.

their entire prey -> their prey whole

"Further" -> "Further,".  The rest of the sentence needs a change too: "they have no aversion to" should be followed by a gerund, but "eat carrion" and "commit cannibalism" aren't gerunds. Maybe "Further, they have no aversion to eating carrion and even committing cannibalism, which are both regular sights."

The last sentence is ambiguous, specifically, whether "their own cold blooded nature" refers to their cannibalism or to their body temperature. I assume it means the latter?

**Comment by sigurdfdragon** [comment]
@jostephd thanks, addressed most of your latest comments. Went with something different for their 'entire prey -> their prey whole', as the last part made me think of snakes eating. Not sure what to do with the cold-blooded part yet.

@CelticMinstrel gender=male isn't needed, it's the default if unspecified in unit_type.
I'll get to your suggestions in the next day or two.

@sevu I can see where something that was gender-specific would cause OOS if one had 1.14.4 and the gender change was in 1.14.5. Unless we've missed something, I'll only apply that to master branch.

**Comment by sigurdfdragon** [comment]
Still not sure what to do about the cold-blooded comment. Would it be a problem to leave it ambiguous?

**Comment by jostephd** [comment]
@sigurdfdragon Thanks for all the changes so far. I think some changes are still needed: the expression "their own cold-blooded nature" refers to the "Saurian males", not to Saurians in general. Changing "their own" to "the Saurians'" and "nature" to "physiology" would resolve the ambiguity.

If you prefer a slightly larger change to that sentence, maybe "The males of the Saurian race have adapted to cannibalism and inter-clan fighting by specializing in cold magics to take advantage of the Saurians’ cold-blooded physiology.". But just the two changes from the previous paragraph would be enough too IMHO.

Also, "warriors;" -> "warriors:" in the 4th paragraph.

**Comment by sigurdfdragon** [comment]
Dropped the cold-blooded part.
Seems slightly redundant, as it mentions the males practice magic earlier.
Also, I'd rather not take a position on if they are warm or cold-blooded.

I'll merge in a few more days if there are no suggestions.

**Comment by jostephd** [comment]
@sigurdfdragon LGTM

**Comment by jostephd** [comment]
One last thing:

> Within each gender the members compete; and through skill, determination, and reputation establish a clear pecking order, with a chief at the top.

The part after the semicolon has a verb but no subject. Possible fixes: change the semicolon to a comma; or delete "and" and add "they" before "establish";

**Comment by CelticMinstrel** [comment]
There is a subject there, actually - it's "members".

**Comment by CelticMinstrel** [comment]
I'm saying I read it as both parts sharing the same subject.

**Comment by sigurdfdragon** [comment]
I changed the semicolon to a comma, seemed better that way.

File: data/core/units.cfg [hunk 7]
```diff
@@ -247,7 +247,24 @@ The Dunefolk’s inquisitive and explorative nature does not preclude military s
         male_name= _ "race^Saurian"
         female_name= _ "race+female^Saurian"
         plural_name= _ "race^Saurians"
-        description= "" # wmllint: ignore
+        description= "Saurians are lizard-like creatures. Smaller and more slender than humans, they rarely stand taller than a ten year old child, though from tip of snout to end of tail a Saurian can be as long as the average man is tall. Light and nimble, the warriors prefer to fight as they hunt — slipping through enemy lines to target the weak and the injured while evading their attackers.
```
Comment (stevecotton) [inline data/core/units.cfg]: This should be a translatable string (only saw this once it was already merged).

Comment (sigurdfdragon) [reply data/core/units.cfg]: Drat. Thanks.

**Pull request merged**